frontend/service/Services.ts

```diff
@@ -109,27 +109,8 @@ const updateService = async ({
   useMechMarketplace?: boolean;
 }): Promise<MiddlewareServiceResponse> =>
   fetch(`${BACKEND_URL_V2}/service/${serviceConfigId}`, {
-    method: 'PUT',
-    body: JSON.stringify({
-      ...serviceTemplate,
-      deploy,
-      configurations: {
-        ...serviceTemplate.configurations,
-        // overwrite defaults with chain-specific configurations
-        ...Object.entries(serviceTemplate.configurations).reduce(
-          (acc, [middlewareChain, config]) => {
-            acc[middlewareChain] = {
-              ...config,
-              rpc: CHAIN_CONFIG[asEvmChainId(middlewareChain)].rpc,
-              staking_program_id: stakingProgramId,
-              use_mech_marketplace: useMechMarketplace,
-            };
-            return acc;
-          },
-          {} as typeof serviceTemplate.configurations,
-        ),
-      },
-    }),
+    method: 'PATCH',
+    body: JSON.stringify({ hash: serviceTemplate.hash }),
```

Comment:
Don't pass the full template here! a PATCH Method will override all passed values here, including the stored environment variables on the backend. See the example [here](https://github.com/valory-xyz/olas-operate-app/pull/621/files#diff-3f17a22cb6687abb1ec7c36c7dbb9d7463fd1f2c7680ba28db62d1f7f93b5b41).